### ISSUE TITLE
Support date formats with dot separator such as DD.MM.YYYY

### ISSUE THREAD

**Issue by N3on1x** [xuri/nfp#7]
**Is your feature request related to a problem? Please describe.**
When using [excelize](https://github.com/qax-os/excelize) to read an Excel workbook with dates formatted as DD.MM.YYYY it results in `..YYYY`, e.g. `18.11.2023` is parsed as `..2023`.

**Describe the solution you'd like**
Once the parser sees the dot `.`, it interprets this as a `TokenTypeDecimal` and discards the preceding datetime values, e.g. parsing `MM.` results in a `.` token.  
I want to see an `MM` token of `TokenTypeDatetime` and a `.` of type `TokenTypeLiteral` being added to the tokens list.

**Describe alternatives you've considered**

**Additional context**

**Comment by xuri** [comment]
Thanks for your issue. I have tested number format code `DD.MM.YYYY` with the excelize master branch, and it works well. Please try to upgrade master branch code, if still have this problem, please let me know.

**Comment by N3on1x** [comment]
Sorry for not adding more context. Let me rectify:

The problem occurs when the datetime is stored as a float in the specific cell rather than a shared string, and a custom format code is used. See this example:

```go
package main

import (
	"fmt"
	"log"

	"github.com/xuri/excelize/v2"
)

func main() {
	f := excelize.NewFile()
	defer func() {
		if err := f.Close(); err != nil {
			log.Fatal(err)
		}
	}()
	activeSheetName := f.GetSheetName(f.GetActiveSheetIndex())
	if err := f.SetCellValue(activeSheetName, "A1", 45186.0); err != nil {
		log.Fatal(err)
	}
	exp := "DD.MM.YYYY"
	style, err := f.NewStyle(&excelize.Style{CustomNumFmt: &exp})
	if err != nil {
		log.Fatal(err)
	}
	if err := f.SetColStyle(activeSheetName, "A", style); err != nil {
		log.Fatal(err)
	}
	val, _ := f.GetCellValue(activeSheetName, "A1")
	fmt.Println(val) // Getting '..2023'. Want '17.09.2023'
}

```

During execution, the circumstances lead to [this line](https://github.com/qax-os/excelize/blob/744236b4b840d71cff471e45f8641d0a682d0292/cell.go#L603) in the excelize codebase to be run, which ultimately lead to `nfp.format()` to be called.

I have addressed the issue in #8
